scipy/integrate/_tanhsinh.py

```diff
@@ -1077,12 +1076,20 @@ def nsum(f, a, b, *, step=1, args=(), log=False, maxterms=int(2**20), tolerances
     that appear in the sum has little effect. If there is not a natural
     extension of the function to all reals, consider using linear interpolation,
     which is easy to evaluate and preserves monotonicity.
-    
+
     The approach described above is generalized for non-unit
     `step` and finite `b` that is too large for direct evaluation of the sum,
-    i.e. ``b - a + 1 > maxterms``.
+    i.e. ``b - a + 1 > maxterms``. It is further generalized to unimodal
+    functions by directly summing terms surrounding the maximum.
+    This strategy may fail:
+
+    - If the left limit is finite and the maximum is far from it.
+    - If the right limit is finite and the maximum is far from it.
+    - If both limits are finite and the maximum is far from the origin.
```

Comment:
Do you think it make sense to allow passing in a guess for the maximum? A user could even find it with `bracket_minimum` first and pass it to `nsum`.

Reply:
Yeah, I thought about that because I thought we might like to use the mode of the distribution as the origin. I figured that the solution for now could be for the user to shift their origin, at least in the doubly infinite case. Follow-up PR, if that sounds good? I'd also like to get this array API compatible before the release.

Reply:
Yes, I think it's fine for a follow up. We could either implement it, or document that the user should shift the origin.